_posts/2023-07-06-optimize-opensearch-index-shard-size.markdown

```diff
@@ -0,0 +1,109 @@
+
+<img src="/assets/media/blog-images/2023-07-06-optimize-index-shard-size/cat_shards.png" alt="cat/indicies"/>{: .img-fluid }
+
+## Change the number of primary shards
+It is always best to configure an appropriate number of primary shards when an index is created. The number of primary shards cannot be easily adjusted after an index is created. To adjust the number of primary shards, you must create a new index with the desired number of shards and then either use the reindex API to copy data from the old index to the new index or reingest the data into the new index from the source.
+
+Following along with the example, the following API call creates a new index with one primary shard and one replicate.
+
+```
+PUT sample-data-1-1
+{
+  "settings": {
+  "index": {
+    "number_of_shards": 1,
+    "number_of_replicas": 1
+  }
+  }
+}
+```
+
+Once the new index is created with the desired number of primary shards, the reindex API can copy data to the new index from the old index.
+
+Following along with the example, the following API call copies data from the *sample-data-5-1* index to the *sample-data-1-1* index.
+
+```
+POST _reindex
+{
+  "source": {
+    "index": "sample-data-5-1"
+  },
+  "dest": {
+    "index": "sample-data-1-1"
+  }
+}
+```
+
+Note a few additional considerations:
+* Before reindexing or copying data from the old index to the new index you may consider setting the old index to read only. This ensures that no new data is written to the old index during the copy.
+* Free storage space should be considered before reindexing. When copying data to a new index, the new index will consume storage. Ensure you have sufficient free storage before reindexing.
+* When creating a new document, always consider using index templates if available. Index templates contain schema information for an index including data types. Using an index template during index creation helps ensure the correct settings are applied to the new index. Index templates are often used in log analytics workloads where new indexes are created frequently.
+
+Following along with the example, we could run the following API call to block all incoming writes to the *sample-data-5-1* index.
+
+```
+PUT sample-data-5-1/_settings
+{
+  "index": {
+    "blocks.read_only": true
+  }
+}
+```
+
+## Change the number of replica shards
+The number of replicate shards for an index can be easily adjusted after an index is created. The index setting API allows for changing the number of replica shards without needing to create a new index and copying data.
+
+An OpenSearch index has both dynamic and static settings. You can change dynamic index settings at any time, but static settings cannot be changed after index creation. The number of replicas is a dynamic setting, hence why it can be changed without needing to create a new index and copying the data.
```

Comment:
```suggestion
An OpenSearch index has both dynamic and static settings. You can change dynamic index settings at any time, but static settings cannot be changed after index creation. The number of replicas is a dynamic setting, hence why it can be changed without needing to create a new index and copy the data.
```